src/main/java/org/folio/circulation/domain/LoanRepository.java

```diff
@@ -312,11 +314,10 @@ private static void keepPatronGroupIdAtCheckoutProperties(Loan loan, JsonObject
       return completedFuture(succeeded(multipleRequests));
     }
 
-    final Result<CqlQuery> statusQuery = getStatusCQLQuery("Open");
-    final Result<CqlQuery> itemIdQuery = exactMatchAny(ITEM_ID, itemsToFetchLoansFor);
+    final FindWithMultipleCqlIndexValues<Loan> fetcher =
+      findWithMultipleCqlIndexValues(loansStorageClient, RECORDS_PROPERTY_NAME, Loan::from);
 
-    return queryLoanStorage(statusQuery.combine(
-        itemIdQuery, CqlQuery::and), PageLimit.limit(requests.size()))
+    return fetcher.findByIdIndexAndQuery(itemsToFetchLoansFor, ITEM_ID, getStatusCQLQuery("Open"))
```

Comment:
Was this the issue, that it was using an older way of fetching the loans?

Reply:
Yes, it was. Here we use the old way of fetching multiple elements.